src/sinks/doris/service.rs

```diff
@@ -0,0 +1,151 @@
+                    response_json.to_string()
+                }
+            };
+
+            info!(
+                message = "Doris stream load response received.",
+                status_code = %http_status_code,
+                stream_load_status = %stream_load_status,
+                response = %formatted_json,
+                internal_log_rate_limit = true
+            );
+        }
+        if http_status_code.is_success() && stream_load_status == StreamLoadStatus::Successful {
+            // Emit metrics for successfully loaded data
+            let load_bytes = response_json
+                .get("LoadBytes")
+                .and_then(|b| b.as_i64())
+                .unwrap_or(0);
+            let loaded_rows = response_json
+                .get("NumberLoadedRows")
+                .and_then(|r| r.as_i64())
+                .unwrap_or(0);
+
+            if loaded_rows > 0 || load_bytes > 0 {
+                emit!(DorisRowsLoaded {
+                    loaded_rows,
+                    load_bytes,
+                });
+            }
+
+            // Emit metrics for filtered rows
+            if let Some(filtered_rows) = response_json
+                .get("NumberFilteredRows")
+                .and_then(|r| r.as_i64())
+                && filtered_rows > 0
+            {
+                emit!(DorisRowsFiltered { filtered_rows });
+            }
+        }
+        Ok(())
+    }
+}
+#[derive(Debug, Snafu)]
+pub struct DorisResponse {
+    pub metadata: RequestMetadata,
+    pub http_response: Response<Bytes>,
+    pub event_status: EventStatus,
+}
+
+impl DriverResponse for DorisResponse {
+    fn event_status(&self) -> EventStatus {
+        self.event_status
+    }
+
+    fn events_sent(&self) -> &GroupedCountByteSize {
+        self.metadata.events_estimated_json_encoded_byte_size()
+    }
+
+    fn bytes_sent(&self) -> Option<usize> {
+        Some(self.metadata.request_encoded_size())
+    }
+}
+
+impl Service<HttpRequest<DorisPartitionKey>> for DorisService {
+    type Response = DorisResponse;
+    type Error = crate::Error;
+    type Future = BoxFuture<'static, Result<Self::Response, Self::Error>>;
+
+    fn poll_ready(&mut self, _cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
+    }
+
+    fn call(&mut self, req: HttpRequest<DorisPartitionKey>) -> Self::Future {
+        let service = self.clone();
+
+        let future = async move {
+            let mut request = req;
+            let table = request.get_additional_metadata().table.clone();
+            let database = request.get_additional_metadata().database.clone();
+            let doris_response = service
+                .client
+                .send_stream_load(database, table, request.take_payload())
+                .await?;
+            let report_response = doris_response.clone();
+            let _ = service.reporter_run(report_response).await;
```

Comment:
Why is the error silently ignored here? The future returns `Result<Self::Response, Self::Error>` and `Self::Error` has the same is the same error returned by `reporter_run`